packages/relay/tests/lib/eth/eth_getBlockByNumber.spec.ts

```diff
@@ -568,34 +568,36 @@ describe('@ethGetBlockByNumber using MirrorNode', async function () {
   });
 
   [false, true].forEach((showDetails) => {
-    it(`eth_getBlockByNumber should skip wrong nonce transactions when showDetails = ${showDetails}`, async () => {
-      // mirror node request mocks
-      restMock.onGet(`blocks/${BLOCK_NUMBER}`).reply(200, DEFAULT_BLOCK);
-      restMock.onGet(BLOCKS_LIMIT_ORDER_URL).reply(200, MOST_RECENT_BLOCK);
-      restMock.onGet(CONTRACT_RESULTS_WITH_FILTER_URL).reply(200, {
-        results: [
-          ...defaultContractResults.results,
-          { ...defaultContractResults.results[0], result: 'WRONG_NONCE' },
-          { ...defaultContractResults.results[0], error_message: prepend0x(ASCIIToHex('WRONG_NONCE')) },
-        ],
+    ['WRONG_NONCE', 'INVALID_ACCOUNT_ID'].forEach((status) => {
+      it(`eth_getBlockByNumber should skip ${status} transactions when showDetails = ${showDetails}`, async () => {
+        // mirror node request mocks
+        restMock.onGet(`blocks/${BLOCK_NUMBER}`).reply(200, DEFAULT_BLOCK);
+        restMock.onGet(BLOCKS_LIMIT_ORDER_URL).reply(200, MOST_RECENT_BLOCK);
+        restMock.onGet(CONTRACT_RESULTS_WITH_FILTER_URL).reply(200, {
+          results: [
+            ...defaultContractResults.results,
+            { ...defaultContractResults.results[0], result: status },
+            { ...defaultContractResults.results[0], error_message: prepend0x(ASCIIToHex(status)) },
+          ],
+        });
+        restMock.onGet(CONTRACT_RESULTS_LOGS_WITH_FILTER_URL).reply(200, DEFAULT_ETH_GET_BLOCK_BY_LOGS);
+
+        const result = await ethImpl.getBlockByNumber(numberTo0x(BLOCK_NUMBER), showDetails, requestDetails);
+
+        RelayAssertions.assertBlock(
+          result,
+          {
+            hash: BLOCK_HASH_TRIMMED,
+            gasUsed: TOTAL_GAS_USED,
+            number: BLOCK_NUMBER_HEX,
+            parentHash: BLOCK_HASH_PREV_TRIMMED,
+            timestamp: BLOCK_TIMESTAMP_HEX,
+            transactions: [CONTRACT_HASH_1, CONTRACT_HASH_2], // should not include the transaction with wrong nonce
```

Comment:
Q: what about invalid account id. How does this confirm that transaction is excluded as it looks exactly like the other test cases that exclude only the wrong nonce

Reply:
[Here](https://github.com/hashgraph/hedera-json-rpc-relay/pull/3177/files/d51906635d7caa3b6dc4a5b8df67e259e5c2faa8#diff-fb9c67af94139d95a8370c6992ae37554d06c8b7490fce0368e6719c49856302R571) we are generating 4 tests for each `WRONG_NONCE`/`INVALID_ACCOUNT_ID` and showDetails `true/false`.

[Here](https://github.com/hashgraph/hedera-json-rpc-relay/pull/3177/files/d51906635d7caa3b6dc4a5b8df67e259e5c2faa8#diff-fb9c67af94139d95a8370c6992ae37554d06c8b7490fce0368e6719c49856302R577) we are returning:
- defaultContractResults
- transaction with results `WRONG_NONCE` or `INVALID_ACCOUNT_ID` (depends on the iteration above)
- transaction with error_message `WRONG_NONCE` or `INVALID_ACCOUNT_ID` again depends on the iteration above

And finally, in the assertions, we check that the response contains only `defaultContractResults` and there is no tx reverted by hedera-specific validation.

PS: I edited the test to make it more understandable.